honeybadgermpc/betterpairing.py

```diff
@@ -361,27 +496,25 @@ def __add__(self, other):
             return ZR(out)
         else:
             raise TypeError(
-                'Invalid addition param. Expected ZR or int. Got '
-                + str(type(other)))
+                'Invalid addition param. Expected ZR or int. Got ' + str(type(other)))
 
     def __iadd__(self, other):
+        self.pp = []
         if type(other) is ZR:
             self.val.add_assign(other.val)
             return self
         elif type(other) is int:
             if other < 0:
-                other *= 1
+                other *= -1
```

Comment:
I think this is a bug.

Reply:
`other *= -1` can be different than `other = other * -1`. 
The difference is `*=` applied to an object actually runs the `__imul__` operator, which for the `ZR` `G1` etc objects we have here, performs modifications in place. I suspect this is the cause of problems